utils/task_tester/inginious-test-task

```diff
@@ -0,0 +1,316 @@
+    print(task.get_response_type())
+    if "text" in job_result:
+        job_result["text"] = ParsableText(job_result["text"], task.get_response_type()).parse()
+    """
+    if "problems" in job_result:
+        for problem in job_result["problems"]:
+            status, log = job_result["problems"][problem]
+            print(ParsableText(log, task.get_response_type()))
+            job_result["problems"][problem] = (status, ParsableText(log, task.get_response_type()).parse())
+    """
+
+def test_task(course, taskid):
+
+    # Get task from its id
+    task = course.get_task(taskid)
+
+    # Build test directory path for current task
+    test_dir = os.path.join(course.get_fs().prefix, taskid, 'test/')
+
+    # List sample submissions for the current task
+    inputfiles = glob.glob(test_dir + '*.test')
+
+    print("\033[34;1m-> Re-running submissions for task <%s> \033[0m" % taskid)
+
+    # For each submission in the test directory, test the task with the specified input
+    for filename in inputfiles:
+        # Open the input file and merge with limits
+        try:
+            inputfile = open(filename, 'r')
+        except IOError as e:
+            print(e)
+            exit(2)
+
+        data = inginious.common.custom_yaml.load(inputfile)
+        launch_job(filename, data, inputfiles, task)
+    
+    result = (job_done_callback.failed, job_done_callback.jobs_done)
+
+    if len(job_done_callback.failed) > 0:
+        print("\033[31;1m%i tests failed\n%s\033[0m" % (len(job_done_callback.failed), '\n'.join(['- %s' % i for i in job_done_callback.failed])) )
+    else:
+        if job_done_callback.jobs_done == 0:
+            print("\033[36;1mNo test submission found \033[0m")
+        else:
+            print("\033[32;1mAll the previous submissions passed \033[0m")
+
+    job_done_callback.failed = []
+    job_done_callback.jobs_done = 0
+
+    return result
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("courseid", help="Course id of the task to test")
+    parser.add_argument("taskids", nargs='*', help="Task id of the task to test")
+    parser.add_argument("-c", "--config", help="Configuration file", default="")
+    parser.add_argument("-v", "--verbose", help="Display more output", action='store_true')
+    parser.add_argument("-p", "--plugins", nargs="*", help="Additional plugins to load")
+    args = parser.parse_args()
+
+    # Read input argument
+    verbose = args.verbose
+    courseid = args.courseid
+    taskids = args.taskids
+    plugins = args.plugins
+
+    # Parse course configuration
+    config = get_config(args.config)
+
+    # Initialize course/task factory
+    task_directory = config["tasks_directory"]
+
+    task_dispensers = {TableOfContents.get_id(): TableOfContents, CombinatoryTest.get_id(): CombinatoryTest}
+
+    # Set basic problem types available
+    # TODO: get that somewhere rather than hardcoding that in the test script
```

Comment:
Maybe this require some refactor of the base code, as the following 10+ lines loading the plugins are, I guess, copy-pasted from it or at least do the same job.

Reply:
This is discussed here https://github.com/UCL-INGI/INGInious/issues/864.

Reply:
If https://github.com/UCL-INGI/INGInious/pull/866 is merged before this one, I will directly use the auto-discovery functions here.

Reply:
Done in ce34c1f73c4cf901679a5597690f3894858d42a0.